### PR TITLE
Expect flat error response from sqs and standardize Forage Error

### PR DESCRIPTION
**Pull request by dleis612**
# Description
Updates were made to expect the new error structure received from the Forage API and make the error responses uniform throughout the SDK. This change was required to match other changes throughout the Forage ecosystem and to improve the developer experience.

## Tests Added / Updated?
Yes, tests were updated. All tests regarding error scenarios needed to be changed to check for the new error pattern.

### CODE REVIEW

File: forage-android/src/main/java/com/joinforage/forage/android/network/model/ForageApiResponse.kt [hunk 1]
```diff
@@ -1,7 +1,74 @@
 package com.joinforage.forage.android.network.model
 
+import org.json.JSONObject
+
 sealed class ForageApiResponse<out T> {
     data class Success<out T>(val data: T) : ForageApiResponse<T>()
 
-    data class Failure(val message: String) : ForageApiResponse<Nothing>()
+    data class Failure(val httpStatusCode: Int, val errors: List<ForageError>) : ForageApiResponse<Nothing>()
```
Comment (ForageRob) [inline forage-android/src/main/java/com/joinforage/forage/android/network/model/ForageApiResponse.kt:8]: Any reason to have `httpStatusCode` here if it is included on the `ForageError` objects?

File: forage-android/src/main/java/com/joinforage/forage/android/ForageSDK.kt [hunk 2]
```diff
@@ -43,7 +44,7 @@ object ForageSDK : ForageSDKApi {
             ).tokenizeCard(
                 cardNumber = currentEntry.getPanNumber()
             )
-            else -> ForageApiResponse.Failure("Invalid PAN entry")
+            else -> ForageApiResponse.Failure(400, listOf(ForageError(400, "bad_request", "Invalid PAN entry")))
```
Comment (ForageRob) [inline forage-android/src/main/java/com/joinforage/forage/android/ForageSDK.kt:47]: "bad_request" is not a valid code from [this list](https://docs.joinforage.app/docs/error-codes). Let's try to be close to that or at least use something from [the backend list](https://github.com/teamforage/payments-api/blob/854844c3a97740d4e32491d5ddf8e29e4006ecb5/api/error_response.py#L162).

Comment (ForageRob) [reply forage-android/src/main/java/com/joinforage/forage/android/ForageSDK.kt:47]: This comment applies in numerous places where you manually construct a `ForageApiResponse.Failure`

File: forage-android/src/main/java/com/joinforage/forage/android/collect/VGSPinCollector.kt [hunk 3]
```diff
@@ -89,10 +90,14 @@ internal class VGSPinCollector(
                     )
                     is VGSResponse.ErrorResponse -> continuation.resumeWith(
                         Result.success(
-                            ForageApiResponse.Failure(response.body.orEmpty())
+                            ForageApiResponse.Failure(response.errorCode, listOf(ForageError(response.errorCode, "user_error", "Invalid Data")))
+                        )
+                    )
+                    null -> continuation.resumeWith(
+                        Result.success(
+                            ForageApiResponse.Failure(500, listOf(ForageError(500, "server_error", "Unknown Server Error")))
```
Comment (ForageRob) [inline forage-android/src/main/java/com/joinforage/forage/android/collect/VGSPinCollector.kt:98]: `unknown_server_error` would be a valid code from our backend

**Review by ForageRob — CHANGES_REQUESTED** [review]
I think the final error format is slightly off what we agreed.

Also does iOS sample app need updating?

File: forage-android/src/main/java/com/joinforage/forage/android/collect/VGSPinCollector.kt (hunk 3 above)
Comment (dleis612) [reply forage-android/src/main/java/com/joinforage/forage/android/collect/VGSPinCollector.kt]: All error codes have been updated here: [7b8dd2b](https://github.com/teamforage/forage-android-sdk/pull/23/commits/7b8dd2baa3cb17edf7cf883683e0fc21e3f2c654)

File: forage-android/src/main/java/com/joinforage/forage/android/network/model/ForageApiResponse.kt (hunk 1 above)
Comment (dleis612) [reply forage-android/src/main/java/com/joinforage/forage/android/network/model/ForageApiResponse.kt]: Updated form of error here: [522c7d8](https://github.com/teamforage/forage-android-sdk/pull/23/commits/522c7d8102efadda9db0bc19cd411cb5abca26bd) to match previously discussed.

**Review by djoksimo — APPROVED** [review]
LGTM 🚀

**Review by ForageRob — APPROVED** [review]

**Pull request merged**